src/main/webapp/WEB-INF/templates/search.jsp

```diff
@@ -17,17 +19,28 @@
       </div>
       <div class="card-container">
         <c:forEach items='${requestScope["cardList"]}' var="knowledgeCard">
-          <div class="card">
-            <header>
-              <img class="icon" src="${knowledgeCard.getIcon()}" alt="${knowledgeCard.getSource()} site icon" width="25px" height="25px">
-              <a href="${knowledgeCard.getLink()}" class="card-title">
-                ${knowledgeCard.getTitle()}
-              </a>
-            </header>
-            <c:if test="${knowledgeCard.getCode() != null}">
-              <pre><code>${knowledgeCard.getCode()}</code></pre>
-            </c:if>
-            <p>${knowledgeCard.getDescription()}</p>
+          <div class="card" url="${knowledgeCard.getLink()}">
```

Comment:
url is not a valid HTML attribute. What you're looking for is https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes